### ISSUE TITLE
DDL - Download of series zip file (multiple issues in one zip)

### ISSUE THREAD

**Issue by jakem742** [mylar3/mylar3#1144]
Many older series are included on GetComics as a zip containing the full series + annuals. In the past, I have assumed there is no alternative than to (1) manually download these (2) drop the files into the series folder (3) manually post-process (name + tag). Just today for the first time, I've noticed that some series will actually download as a zip using DDL and automatically process the contents. 

I have since been trying to figure out how to force Mylar to recognise zips of other series on the site, but can't figure it out! I have tried changing the series name to match the GetComics DL title, however that didn't help. Are there any fields I can tweak to help Mylar recognise these series zips?

tl;dr - I'm not sure how to manually get Mylar to recognise series that are in zip format (full volume in one file). eg. [Mighty Avengers (2013)](https://getcomics.info/marvel/mighty-avengers-001-014-free-download/) Is there anything I can tweak to make these files download correctly?

My apologies if this is incorrectly marked as a "bug", however I figured it fits better here than in "feature requests"?

**Comment by evilhero** [comment]
There is an ``allow_packs`` option in the ini that will allow Mylar to attempt to use packs that are returned in a search response, but alot depends on how GC has named it. Setting that to ``True`` will allow it to accept packs as a valid result, download it, extract it and then post-process them accordingly. It will mark each issue in the pack as Snatched during the initial snatch of the file too.

There is also a ``pack_priority`` setting in the config.ini that was recently added. It will prioritise pack downloads over single issues if the value is set to ``True``, if it's set to ``False`` (which is the default setting), singles will get matched against first and if it can't find a match in the singles it will look to a pack.

But again it comes down to how GC names their packs - if it's just like the one you linked above, that should be enough to allow Mylar to match up and download it. But those that have extra things in them (+TPBs - Extra content - Fan Filled - w/e) makes it harder to parse cleanly and to err on the side of caution it won't download those as it can't be sure exactly what the match includes.

I've got a few fixes for the packs option for GC coming, which hopefully will allow for some better matching of packs as opposed to rejecting them - but it needs to be tested still unfortunately.

**Comment by jakem742** [comment]
Can I just say, this is an amazing feature and it will be life-changing once you've got it all finished up! :) 

While I understand the need to get exact terms in the search, I imagine it isn't too difficult to make exceptions for annuals and TPB's in the pack and simply delete any leftover files from the cache once everything is processed? It seems a shame to reject results simply because the include annuals. I imagine it gets a little more tricky when a single page has multiple download links on it though...

I have just checked those config options and noticed that both allow_packs and pack_priority are "False" (I've now set them to true). I am assuming that "false" is the default at the minute. Is it then a bug if I have had several zip files downloaded while the config allow_packs is false?

I did some research on what is happening with the "Mighty Avengers (2013)" series. When I click the "Search-4-Missing" button on the series page the log includes the following snippet:

```
2022-04-06 15:52:25 | DEBUG | Siege - Mighty Avengers (TPB) (2011) [155 M]
-- | -- | --
2022-04-06 15:52:25 | DEBUG | Avengers - Mighty Origins (TPB) (2018) [206 M]
2022-04-06 15:52:25 | DEBUG | Mighty Avengers Vol.  (TPB) #1 [534 M]
2022-04-06 15:52:25 | DEBUG | Mighty Avengers Vol. 1 #1 [1.2 M]
2022-04-06 15:52:25 | DEBUG | Thor - The Mighty Avenger #1 [284 M]
2022-04-06 15:52:25 | DEBUG | Thor - The Mighty Avenger - The Complete Collection (2013) [309 M]
2022-04-06 15:52:25 | DEBUG | Mighty Avengers by Dan Slott - The Complete Collection (2019) [697 M]
2022-04-06 15:52:25 | DEBUG | Mighty Avengers by Brian Michael Bendis - The Complete Collection (2017) [760 M]
2022-04-06 15:52:25 | DEBUG | Captain America - Fighting Avenger #1 (2011) [111 M]
2022-04-06 15:52:25 | DEBUG | Avengers - Nights of Wundagore (TPB) (2015) [90 M]
2022-04-06 15:52:25 | DEBUG | Captain America and the Mighty Avengers + TPBs #1 [822 M]
2022-04-06 15:52:25 | DEBUG | The Mighty Avengers - The Unspoken (TPB) (2010) [141 M]
2022-04-06 15:52:25 | DEBUG | pagelines: [2]
2022-04-06 15:52:25 | INFO | There are 12 results
2022-04-06 15:52:24 | DEBUG | [PROVIDER-SEARCH-DELAY][DDL] Last search took place 209 seconds ago. We're clear...
2022-04-06 15:52:24 | DEBUG | [DDL-QUERY] Query set to: Mighty Avengers 2013
2022-04-06 15:52:24 | DEBUG | removing query from loop that accounts for no issue number
2022-04-06 15:52:24 | DEBUG | Sending request to DDL(GetComics) site for : Mighty Avengers 2
2022-04-06 15:52:24 | DEBUG | findloop: 0 / findcount: 1
2022-04-06 15:52:24 | INFO | Shhh be very quiet...I'm looking for Mighty Avengers issue: 2 (2013) using DDL(GetComics).

[2](https://getcomics.info/page/2/?s=Mighty+Avengers+2013)
```

Based on the log entries above, I'm wondering if the search term is getting messed up somehow? The query listed as [2] ("Mighty Avengers 2013") only has one result (the correct one), however the actual search mylar runs is returning results for the search term ("Mighty Avengers"), leading to the 12 results listed (which don't include the correct one in the first page for some reason). I have included the full care package in case it's more helpful.

Other than that, I guess I'll just eagerly wait for the next few updates to drop. :) I assume it is worth jumping on to the py3-dev build for features like this? You indicated it should be fairly stable in the FAQ.
